sync_labels.py

```diff
@@ -45,35 +45,44 @@ def sync_labels(src_repo, dest_repo, dry_run, delete):
             print(f" [-] {lab}")
 
     # Add new labels
+    add_ok = []
+    add_fail = []
     for name in dest_add:
-        print(f"Creating label {name} in destination repo {dest_repo}...")
         if dry_run:
-            print(f"Dry run successful!")
+            add_ok.append(name)
             continue
         label = s.get_label(name)
         color = label.color
         description = label.description
         try:
             d.create_label(name, color, description)
```

Comment:
Consider removing the `dry_run` conditional block above and instead doing:
```suggestion
            if not dry_run:
                d.create_label(name, color, description)
```
This way the `dry_run` and "wet" run code paths share as much code as possible. Aiming for that would help prevent unexpected changes when switching between them.

Reply:
This is a good suggestion, but carried out, might require more lines. The label, color, description variables, and the create_label call, are all part of the wet path, so wrapping all calls in an `if not dry_run` means the try/except/else is in the wet path.

Compare the as-is version (including the `else` part of the try/except/else per your comment above):

```
    for name in dest_add:
        if dry_run:
            add_ok.append(name)
            continue
        label = s.get_label(name)
        color = label.color
        description = label.description
        try:
            d.create_label(name, color, description)
        except GithubException:
            add_fail.append(name)
        else:
            add_ok.append(name)
```

and separating the dry/wet paths:

```
    for name in dest_add:
        if not dry_run:
            label = s.get_label(name)
            color = label.color
            description = label.description
            try:
                d.create_label(name, color, description)
            except GithubException:
                add_fail.append(name)
            else:
                add_ok.append(name)
        else:
            add_ok.append(name)
```

Reply:
I see. Yes I was imaging the that label, color, and description variables were part of the dry path too.

Of these two code blocks, it looks like they're the same amount of lines. I still prefer the second, but they're functionally the same so :shrug: